commands/template_store_list.go

```diff
@@ -117,25 +129,51 @@ func formatTemplatesOutput(templates []TemplateInfo, verbose bool) string {
 	return buff.String()
 }
 
-func formatBasicOutput(lineWriter *tabwriter.Writer, templates []TemplateInfo) {
-	fmt.Fprintf(lineWriter, "NAME\tSOURCE\tDESCRIPTION\n")
-	for _, template := range templates {
-		fmt.Fprintf(lineWriter, "%s\t%s\t%s\n",
-			template.TemplateName,
-			template.Source,
-			template.Description)
+func formatBasicOutput(lineWriter *tabwriter.Writer, templates []TemplateInfo, platform string) {
+	if platform != allPlatforms {
+		fmt.Fprintf(lineWriter, "NAME\tSOURCE\tDESCRIPTION\n")
+		for _, template := range templates {
```

Comment:
Feels like we should avoid repeated code here by pre-filtering the templates of interest.  So we lift out the two checks and then just output what remains.
```
if platform != allPlatforms {
templates = filterTemplates(platform)
}

fmt.Fprintf(lineWriter, "NAME\tLANGUAGE\tPLATFORM\tSOURCE\tDESCRIPTION\n")
for _, template := range templates {
fmt.Fprintf(lineWriter, "%s\t%s\t%s\t%s\t%s\n",
    template.TemplateName,
    template.Language,
    template.Platform,
    template.Source,
    template.Description)
}
```
WDYT?

Reply:
Depending on what we choose in the check function we might need to handle case in filterTemplates

Reply:
I think this is a great proposal I will implement `filterTemplates` to modify the object to contain only the requested platform if one is requested I will also add tests to the two added functions

Reply:
Implemented `filterTempaltes` and removed code duplication ✔️